src/jit/codegenarmarch.cpp

```diff
@@ -1617,11 +1766,29 @@ void CodeGen::genCallInstruction(GenTreeCall* call)
         }
         else
         {
-            regNumber argReg = curArgTabEntry->regNum;
             genConsumeReg(argNode);
```

Comment:
We can call `CodeGen::genConsumeReg(GenTree* tree)` to consume register for each argument . On other architectures, if argument is multi-reg struct (or long), it will be morphed to `FieldList` and each field has its `PutArgReg` node, and we can call `genConsumeReg()` for each `PutArgReg` node. But we don't morph to `FieldList` all multi-reg struct arg on arm32. For example, some parts of very large size struct can be passed using registers and IMO it makes too complex to morph this struct. So I defined new node type `PutArgSplit`, and it has multi-reg targets with stack slot. I updated this feature similar to multi-reg return.